app/models/openscap_result.rb

```diff
@@ -0,0 +1,68 @@
+require 'openscap'
+require 'openscap/ds/arf'
+require 'openscap/xccdf/benchmark'
+
+class OpenscapResult < ApplicationRecord
+  belongs_to :container_image
+  has_one    :binary_blob,           :dependent => :destroy, :autosave => true, :as => :resource
+  has_many   :openscap_rule_results, :dependent => :destroy, :autosave => true
+
+  OpenSCAP.oscap_init
+
+  def html
+    return unless binary_blob && binary_blob.binary
+    with_oscap_file(binary_blob.binary) do |arf_path|
+      arf = OpenSCAP::DS::Arf.new(arf_path)
+      arf.html
+    end
+  end
+
+  def raw=(value)
+    self.binary_blob = BinaryBlob.new(:name => 'openscap_compliance_arf', :data_type => 'XML')
+    binary_blob.binary = value
+    with_oscap_file(value) do |arf_path|
+      create_results(arf_path)
+    end
+  end
+
+  private
+
+  def create_results(arf_path)
+    results, meta = parse_arf(arf_path)
+
+    results.each do |k, v|
```

Comment:
is `k` a `scap_rule`?  It would be helpful to readers like me to use a more descriptive set of variable names here.